shared/util/tf/tf_parser.go

```diff
@@ -0,0 +1,348 @@
+package tf
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/ext/typeexpr"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/pkg/errors"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/convert"
+)
+
+type TfParser struct {
+}
+
+func NewTfParser() TfParser {
+	return TfParser{}
+}
+
+var moduleBlockSchema = &hcl.BodySchema{
+	Blocks: []hcl.BlockHeaderSchema{
+		{
+			Type:       "module",
+			LabelNames: []string{"name"},
+		},
+	},
+}
+
+var variableBlockSchema = &hcl.BodySchema{
+	Attributes: []hcl.AttributeSchema{
+		{
+			Name: "description",
+		},
+		{
+			Name: "default",
+		},
+		{
+			Name: "type",
+		},
+		{
+			Name: "sensitive",
+		},
+		{
+			Name: "nullable",
+		},
+	},
+	Blocks: []hcl.BlockHeaderSchema{
+		{
+			Type: "validation",
+		},
+	},
+}
+
+var outputBlockSchema = &hcl.BodySchema{
+	Blocks: []hcl.BlockHeaderSchema{
+		{
+			Type:       "output",
+			LabelNames: []string{"name"},
+		},
+	},
+}
+
+func (tf TfParser) ParseServices(dir string) (map[string]bool, error) {
+	var services map[string]bool = map[string]bool{}
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			if info.Name() == ".terraform" || info.Name() == ".git" {
+				return filepath.SkipDir
+			}
+		}
+		if !info.IsDir() && filepath.Ext(path) == ".tf" {
```

Comment:
```suggestion
		if info.IsDir() || filepath.Ext(path) != ".tf" { return nil } 
```

then the rest of the logic. Avoid nesting when possible

https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting